document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
+
+### `level` property { #notification-level }
+
+A `notification` object may contain a property named `level` whose value is one of a fixed set of strings
+that specify the severity level of the notification.
+
+If present, the `level` property shall have one of the following values, with the specified meanings:
+
+* `error`: A serious problem was found.
+
+* `warning`: A problem that is not considered to be serious was found.
+
+* `note`: The notification is purely informational.
+
+If the `level` property is absent, it shall be considered equivalent to the value `warning`.
+
+### `time` property { #notification-time }
+
+A `notification` object may contain a property named `time` whose value is a string
+specifying the date and time at which the analysis tool generated the notification.
+The string shall be in the format specified by (&sect;[#fileFormat-dateTime]).
+
+### `exception` property { #notification-exception }
+
+If the issue is a result of a runtime exception, the `notification` object may contain a property named `exception`
+whose value is an `exception` object (&sect;[#exception]).
+
+If the issue is not the result of a runtime exception, the `exception` property shall be absent.
+
+### `tags` property { #notification-tags }
+
+A `notification` object may contain a property named `tags`, whose value is a tag set.
+
+### `properties` property { #notification-properties }
+
+A `notification` object may contain a property named `properties`, whose value is property bag.
+This allows tools to include information about the encountered condition that is not explicitly specified in the SARIF format.
+
+## `exception` object { #exception }
+
+### General { #exception-General }
+
+An `exception` object describes a runtime exception encountered in the course of executing an analysis tool.
+
+### `type` property { #exception-type }
+
+An `exception` object shall contain a property named `type` whose value is a string containing the fully qualified
+type name of the exception.
+
+~ IsoNumberedExample { @IsoNumberedExample=1 }
+C++: `"std::runtime_error"`
+~
+
+~ IsoNumberedExample
+C#: `"System.ArgumentNullException"`
+~
+
+### `message` property { #exception-message }
+
+An `exception` object shall contain a property named `message` whose value is a string that describes the exception.
+
+### `stack` property { #exception-stack }
+
+An `exception` object may contain a property named `stack` whose value is a `stack` object (&sect;[#stack])
+that describes the sequence of function calls leading to the exception.
+
+### `innerExceptions` property { #exception-innerExceptions }
+
+An `exception` object may contain a property named `innerExceptions` whose value is an array of `exception` objects,
+each of which is considered to be a cause of the containing `exception`.
+
+~ IsoNote
+There is commonly no more than one inner exception.
```

Comment:
Rather than suggest this is a special-case to accommodate .NET, we should note that this exists to accommodate platforms that provide a mechanism for aggregating exceptions, such as the .NET AggregateException type. This is useful functionality